callback.go

```diff
@@ -107,10 +107,24 @@ func (cb *ParticipantCallback) Merge(other *ParticipantCallback) {
 type DisconnectionReason string
 
 const (
-	LeaveRequested DisconnectionReason = "leave requested by room"
-	Failed         DisconnectionReason = "connection to room failed"
+	LeaveRequested  DisconnectionReason = "leave requested by room"
+	UserUnavailable DisconnectionReason = "remote user unavailable"
+	RejectedByUser  DisconnectionReason = "rejected by remote user"
+	Failed          DisconnectionReason = "connection to room failed"
```

Comment:
Would be nice to move protocol `DisconnectReason` -> `string` to common place like `protocol` and use it both here and also in `livekit` server. Comment not related to this PR.

Reply:
Actually this might be the right time. Since I need to update both anyway, I could do it as well.

The only problem is that `OnDisconnected` callback accepts this string type currently and changing it to something else would not be backward-compatible. Will have to add a new callback, probably.

Reply:
Do you know if somebody is telling on that string value? Guess it is impossible to say.

We can keep the same string for the two existing there and add the rest.

Reply:
Oh string type. Got it. Yeah that's messy